File: userland/apps/asemics/dbg.c
#include "asemics.h"
#include <stdarg.h>

void initDbg(void) {
    dbg = fopen("/tmp/asemics.log", "w");
    if (!dbg) die("fopen");
    setvbuf(dbg, NULL, _IOLBF, 0);
    fprintf(dbg, "%s", SCREEN_CLEAR);
    fprintf(dbg, "-- ENTRY --\n");
}

void addDbgLog(const char *fmt, ...) {
    va_list ap;
    va_start(ap, fmt);
    fprintf(dbg, " (%d, %d) MODE=%d\n", E.cursor.rx, E.cursor.y, E.mode);
    vfprintf(dbg, fmt, ap);
    va_end(ap);
}

File: userland/apps/asemics-rs/src/dbg.rs
use std::fs::File;
use std::io::Write;

use crate::consts::SCREEN_CLEAR;
use crate::editor::Editor;
use crate::terminal::die;

pub fn init_dbg() -> File {
    let Ok(mut dbg) = File::create("/tmp/asemics.log") else {
        die("fopen")
    };
    let _ = dbg.write_all(SCREEN_CLEAR);
    let _ = dbg.write_all(b"-- ENTRY --\n");
    dbg
}

impl Editor {
    pub fn add_dbg_log(&mut self, msg: &str) {
        let _ = write!(
            self.dbg,
            " ({}, {}) MODE={}\n{}",
            self.cursor.rx, self.cursor.y, self.mode as i32, msg
        );
    }
}
